gossip/src/cluster_info.rs

```diff
@@ -1345,10 +1376,13 @@ impl ClusterInfo {
             .collect()
     }
 
-    fn is_spy_node(contact_info: &LegacyContactInfo, socket_addr_space: &SocketAddrSpace) -> bool {
-        !ContactInfo::is_valid_address(&contact_info.tpu, socket_addr_space)
-            || !ContactInfo::is_valid_address(&contact_info.gossip, socket_addr_space)
-            || !ContactInfo::is_valid_address(&contact_info.tvu, socket_addr_space)
+    fn is_spy_node(node: &LegacyContactInfo, socket_addr_space: &SocketAddrSpace) -> bool {
+        [node.tpu(), node.gossip(), node.tvu()]
+            .into_iter()
+            .any(|addr| match addr {
+                Ok(addr) => !socket_addr_space.check(&addr),
+                Err(_) => true,
+            })
```

Comment:
```suggestion
            .any(|addr| addr.ok().filter(socket_addr_space.check).is_none())
```

this was cooler when i thought there was a `Result::filter`